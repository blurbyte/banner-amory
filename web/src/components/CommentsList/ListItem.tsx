import * as React from 'react';
import styled from 'styled-components';

const Wrapper = styled.li`
  display: grid;
  grid-template-columns: 15rem 1fr;
  height: 3rem;
  align-items: end;
`;

const Name = styled.span`
  font-weight: 600;
  padding-right: 1rem;
`;

const Message = styled.span`
  font-weight: 400;
`;

type ListItemProps = {
  userName: string;
  message: string;
};

class ListItem extends React.Component<ListItemProps> {
  render() {
    const { userName, message } = this.props;

    return (
      <Wrapper>
        <Name>{userName}</Name>
        <Message>{message}</Message>
      </Wrapper>
    );
  }
}

export default ListItem;
