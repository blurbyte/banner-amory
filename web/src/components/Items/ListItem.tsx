// single item in the grid

import * as React from 'react';
import { Link as BaseLink } from '@reach/router';
import styled from 'styled-components';
import { transparentize } from 'polished';

import { Colors } from '../../styles/colors';
import { ItemBasic as ListItemProps } from '../../types/Item';
import Image from '../Image';

const Item = styled.li`
  position: relative;
  background: ${transparentize(0.9, Colors.blue)};
  z-index: 99;
  user-select: none;
`;

const Link = styled(BaseLink)`
  color: ${Colors.blue};
`;

const Rank = styled.p`
  display: flex;
  align-items: center;
  justify-content: center;
  position: absolute;
  right: 0;
  bottom: 0;
  z-index: 99;
  background: ${Colors.white};
  margin: 0;
  padding: 0;
  width: 2rem;
  height: 2rem;
  font-weight: 600;
`;

class ListItem extends React.Component<ListItemProps> {
  render() {
    const { slug, rank } = this.props;

    return (
      <Item>
        <Link to={`item/${slug}`}>
          <Image slug={slug} />
          <Rank>{rank}</Rank>
        </Link>
      </Item>
    );
  }
}

export default ListItem;
