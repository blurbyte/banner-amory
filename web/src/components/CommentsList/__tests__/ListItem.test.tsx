import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import ListItem from '../ListItem';

test('renders correctly', () => {
  const component = renderer.create(
    <ListItem userName="Taylor Swift" message="Hello, lets have a message here!" />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
