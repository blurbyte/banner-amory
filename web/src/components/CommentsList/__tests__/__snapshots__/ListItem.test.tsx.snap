// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 15rem 1fr;
  height: 3rem;
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
}

.c1 {
  font-weight: 600;
  padding-right: 1rem;
}

.c2 {
  font-weight: 400;
}

<li
  className="c0"
>
  <span
    className="c1"
  >
    Taylor Swift
  </span>
  <span
    className="c2"
  >
    Hello, lets have a message here!
  </span>
</li>
`;
